openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/util/CommonSpeedCalculator.java

```diff
@@ -0,0 +1,43 @@
+package org.heigit.ors.routing.graphhopper.extensions.util;
+
+import com.graphhopper.routing.EdgeKeys;
+import com.graphhopper.routing.profiles.DecimalEncodedValue;
+import com.graphhopper.routing.util.FlagEncoder;
+import com.graphhopper.routing.util.SpeedCalculator;
+import com.graphhopper.util.EdgeIteratorState;
+import org.heigit.ors.routing.graphhopper.extensions.storages.ExpiringSpeedStorage;
+
+/**
+ * Simple SpeedCalculator based on a SpeedStorage.
+ * Default speed of Byte.MIN_VALUE results in original graph speed
+ */
+public class CommonSpeedCalculator implements SpeedCalculator {
```

Comment:
I believe this speed calculator should be refactorer to extend `AbstractAdjustedSpeedCalculator` rather than directly implement `SpeedCalculator`, compare e.g. `MaximumSpeedCalculator` and `ConditionalSpeedCalculator`. Such approach is much more flexible as it facilitates chaining of different speed calculators. Instead of maintaining  `avSpeedEnc` use the super speed calculator  `DefaultSpeedCalculator` to query for the base speed.

Reply:
Okay

Reply:
Done

Reply:
Thanks a lot!